cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -0,0 +1,32 @@
+from random import randint
+
+from cfme.middleware.deployment import MiddlewareDeployment
+from utils import testgen
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, ["hawkular"], scope="function")
```

Comment:
somewhere you should have this:

```python
pytestmark = [pytest.mark.userfixtures('setup_provider')]
```

otherwise the provider that is being tested does not get added in the appliance.